samtranslator/model/eventsources/pull.py

```diff
@@ -121,9 +124,13 @@ def to_cloudformation(self, **kwargs):  # type: ignore[no-untyped-def]
         if self.Stream and not self.StartingPosition:
             raise InvalidEventException(self.relative_id, "StartingPosition is required for Kinesis, DynamoDB and MSK.")
 
+        if self.StartingPositionTimestamp:
+            self.StartingPositionTimestamp = float(self.StartingPositionTimestamp)
```

Comment:
this is a possible source of availability dips. 
- Why we need to type casting when the type is pass through
- What if the customer provided `self.StartingPositionTimestamp` cannot be parsed into float?